ui/src/main/resources/FileManagerCode/RecentFilesMacro.xml

```diff
@@ -317,7 +317,9 @@
 #end
 #if ($i == 0)
   * {{translation key="fileManager.recentFiles.noFiles"/}}
-  * [[{{translation key="fileManager.recentFiles.addFiles"/}}&gt;&gt;FileManager.WebHome||queryString="#?driveNode=WebHome"]]
+  #if($hasEdit)
```

Comment:
``$hasEdit`` variable is set in ``xwikivars.vm`` and it refers to the current document, which in your case is the document that calls the macro. This means that if I create a new page and use the macro in its content then I will get the "Add files" link. You need to test is the current user has the right to create a page in the FileManager space. See the [Security Scripting API](http://extensions.xwiki.org/xwiki/bin/view/Extension/Security+Module#HScriptingAPI28since6.1RC129). With this API you can check if the current user has the right to "edit" a space (specified by the space reference).